src/lib/discord/commands/admin.ts

```diff
@@ -0,0 +1,215 @@
+  body += `#### 🕹️ View the original Discord thread [here](${threadUrl})\n`
+  return body
+}
+
+export const config = new SlashCommandBuilder()
+  .setName('admin')
+  .setDescription('Commands for admins.')
+  .addSubcommand((subcommand) =>
+    subcommand
+      .setName('mirror')
+      .setDescription('Mirror a thread to GitHub Discussions')
+      .addStringOption((option) =>
+        option
+          .setName('repository')
+          .setDescription('The AWS Amplify repository')
+          .setRequired(true)
+          .addChoices(
+            ...[...repositories.keys()].map((r) => ({ name: r, value: r }))
+          )
+      )
+  )
+
+export async function handler(
+  interaction: ChatInputCommandInteraction
+): Promise<InteractionReplyOptions | string> {
+  const channel = interaction.channel as ThreadChannel
+  const messages = await channel?.messages?.fetch()
+  const record = await prisma.question.findUnique({
+    where: { threadId: channel.id },
+  })
+  const firstMessage = await channel.fetchStarterMessage()
+
+  const somethingWentWrongResponse = (message?: string) => {
+    return `🤢 something went wrong${message}`
+  }
+
+  const badAccessMessage = (text: string) => {
+    const embed = new EmbedBuilder()
+    embed.setColor('#ff9900')
+    embed.setDescription(text)
+    return { embeds: [embed], ephemeral: true }
+  }
+
+  if (!channel.isThread() || !isThreadWithinHelpChannel(channel))
+    return badAccessMessage(
+      'This command only works in public threads within help channels.'
+    )
+
+  // check if user is admin
+  let access
+  if (interaction?.user?.id) {
+    try {
+      access = await getUserAccess(interaction.user.id)
+    } catch (error) {
+      console.error('Error getting access', error)
+      return somethingWentWrongResponse(': failed to check user admin status')
+    }
+  }
+  if (!access?.isAdmin)
+    return badAccessMessage('Only admins can use this command.')
+  if (!record || !messages?.size)
+    return somethingWentWrongResponse(': failed to fetch thread messages.')
+
+ // create discussion content
+  const title = record?.title
+  const body = createDiscussionBody(firstMessage, messages, record?.url)
+  let answerContent
+
+  // if the answer is solved create solution content
+  if (record.isSolved) {
+    const answerRecord = await prisma.answer.findUnique({
+      where: { questionId: record?.id },
+    })
+    const messageArray = Array.from(messages.values())
+    const answerMessage =
+      messageArray[
+        messageArray.findIndex(
+          (message: Message) => message.id === answerRecord.id
+        )
+      ]
+    if (answerMessage) answerContent = createAnswer(answerMessage)
+  }
+  const repo = repositories.get(
+    interaction.options.getString('repository') as string
+  )
+
+  if (repo?.discussion) {
+    try {
+      const postResponse = await postDiscussion(
+        repo.id,
+        repo.discussion.id,
+        title,
+        body,
+        record.id
+      )
```

Comment:
can these arguments be merged into a single object? Looking into this function and the GraphQL call I see how these arguments line up with the mutation's inputs, however for readability and general api dx it makes sense to move to a single param with named inputs - which can also enable us to [add descriptions to inputs](https://github.com/aws-amplify/discord-bot/blob/main/cdk/src/components/hey-amplify-app.ts#L24-L28). 

@dnys1 I'll leave a mention here if you'd like to add any additional feedback 🙂